staging/src/k8s.io/apiserver/pkg/admission/plugin/namespace/lifecycle/admission.go

```diff
@@ -48,12 +49,20 @@ const (
 	// 2. It allows time for a namespace creation to distribute to members of a storage cluster,
 	//    so the live lookup has a better chance of succeeding even if it isn't performed against the leader.
 	missingNamespaceWait = 50 * time.Millisecond
+	// featureNodeLease is the name of the feature gate for the Node lease feature (features/issues/589).
+	// we make it a constant here since importing k8s.io/kubernetes/pkg/features is forbidden for vendored
+	// dependencies (it forms a cycle between the core repo and external repos).
+	featureNodeLease = "NodeLease"
 )
 
 // Register registers a plugin
 func Register(plugins *admission.Plugins) {
 	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
-		return NewLifecycle(sets.NewString(metav1.NamespaceDefault, metav1.NamespaceSystem, metav1.NamespacePublic))
+		immortalNamespaces := sets.NewString(metav1.NamespaceDefault, metav1.NamespaceSystem, metav1.NamespacePublic)
+		if utilfeature.DefaultFeatureGate.Enabled(featureNodeLease) {
+			immortalNamespaces.Insert(v1.NamespaceNodeLease)
```

Comment:
having the generic apiserver library be concerned about a node-specific thing isn't proper. the fact that referencing the feature flag introduced a cycle was a good signal of this.

Reply:
I'd actually ask why this new namespace is put in this category. the two existing namespaces are there because they are required for the controller manager (which is what cleans up terminating namespaces) to work. I don't anticipate this new namespace is in this category. deletion would be disruptive, but does not put the cluster in a state it cannot recover from.

Reply:
>the two existing namespaces are there because they are required for the controller manager (which is what cleans up terminating namespaces) to work

Ah, I thought it was to protect any system namespaces that should live for the duration of the cluster, not just bare-minimum ones necessary for controller manager to clean up other terminating namespaces. I can remove NodeLease from this, the controller manager should just re-create it if it gets deleted.